### ISSUE TITLE
pygmt.Figure.subplot: "frame", "sharex", and "sharey" do not work correctly with "pygmt.Figure.set_panel"

### ISSUE THREAD

**Issue by yvonnefroehlich** [GenericMappingTools/pygmt#2365]
Using the `pygmt.Figure.subplot` method the x and y axis labels set via the `sharex` and `sharey` parameters do not always show up.

_First mentioned by @weiji14 in https://github.com/GenericMappingTools/pygmt/pull/2354#pullrequestreview-1294425447_:

Been trying to set a subtitle using `sharex="b+lNormalize"` and `sharey="l+lAzimuth"` but it doesn't show up. Wondering if the `+l` is broken, or if there's some setting overriding the label...

_Original post by @yvonnefroehlich at https://github.com/GenericMappingTools/pygmt/pull/2354#issuecomment-1426970572_:

There seems to be a difference between using `with fig.set_panel(panel=index):` and `panel=True`, which is probably not intended.

**(1) `with fig.set_panel(panel=index):`** -> Does NOT add x and y labels
_Example code_:
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
):

    for i in range(nrows):
        for j in range(ncols):
            index = i * nrows + j

            # does NOT add x and y labels
            with fig.set_panel(panel=index):
                fig.basemap(region=[0, 10, 0, 10], projection="X?")

fig.show()
# fig.savefig(fname="issue_labels_doubleloop.png")
```
_Output figure_:
![issue_labels_doubleloop](https://user-images.githubusercontent.com/94163266/218300269-8bf39983-57e9-496a-91c6-611f351040a3.png)

**(2) `panel=True`** -> DOES add x and y labels
_Example code_:
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
):

    for i in range(nrows):
        for j in range(ncols):
                        
            # DOES add x and y labels
            fig.basemap(region=[0, 10, 0, 10], projection="X?", panel=True)

fig.show()
# fig.savefig(fname="issue_labels_panelTrue.png")
```
_Output figure_:
![issue_labels_panelTrue](https://user-images.githubusercontent.com/94163266/218300281-7eff2ac8-fc15-47b2-8b62-3346a029a6c2.png)

**Comment by yvonnefroehlich** [comment]
This issue is also relevant for the  `frame` parameter and affects annotations, ticks and, gridlines.

**(1) `with fig.set_panel(panel=index):`** -> Does NOT add x and y labels, and does NOT adjust frame settings
_Example code_:
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
    frame="a1f0.5g2",
):

    for i in range(nrows):
        for j in range(ncols):
            index = i * nrows + j

            # does NOT add x and y labels, 
            # does NOT adjust frame settings
            with fig.set_panel(panel=index):
                fig.basemap(region=[0, 10, 0, 10], projection="X?")

fig.show()
# fig.savefig(fname="issue_frame_doubleloop.png")
```
_Output figure_:
![issue_frame_doubleloop](https://user-images.githubusercontent.com/94163266/219853027-b84a0059-30cf-43e3-8384-89d6bcf6f34c.png)


**(2) `panel=True`** -> Does add x and y axis labels and does adjust frame settings
_Example code_:
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
    frame="a1f0.5g2",
):

    for i in range(nrows):
        for j in range(ncols):
                        
            # does add x and y labels, 
            # does adjust frame settings
            fig.basemap(region=[0, 10, 0, 10], projection="X?", panel=True)

fig.show()
# fig.savefig(fname="issue_frame_panelTrue.png")
```
_Output figure_:
![issue_frame_panelTrue](https://user-images.githubusercontent.com/94163266/219853038-6db0f861-3da4-4874-870a-42da1fef7e84.png)

**Comment by yvonnefroehlich** [comment]
The problem is related to using `pygmt.Figure.basemap` directly after `pygmt.Figure.set_panel` (please see case (1a) vs. case (1b)). For this case, it "works" only if the `frame` parameter is set again in the same way (please see case (1c)). So, probably the [default] argument passed through the `frame` parameter of `pygmt.Figure.basemap` is overwriting the settings via the parameters of `pygmt.Figure.subplot`.

**(1a) `pygmt.Figure.set_panel:` + using `pygmt.Figure.basemap`**
_Example code_
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
    frame="a1f0.5g2",
):

    for ip in range(nrows*ncols):

        with fig.set_panel(panel=ip):
            
            fig.basemap(region=[0, 10, 0, 10], projection="X?")

fig.show()
# fig.savefig(fname="issue_frame_panelTrue_basemap.png")

```
_Output figure_
![issue_frame_panelTrue_basemap](https://user-images.githubusercontent.com/94163266/219975314-2d69cf4c-0543-4957-b87e-31c938f937ad.png)


**(1b) `pygmt.Figure.set_panel:` + NOT using `pygmt.Figure.basemap`**
_Example code_
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
    frame="a1f0.5g2",
):

    for ip in range(nrows*ncols):

        with fig.set_panel(panel=ip):
            
            fig.plot(
                region=[0, 10, 0, 10],
                projection="X?",
                x=[3,2,5], # some random data
                y=[6,1,8],
                style="c0.2c",
            )

fig.show()
# fig.savefig(fname="issue_frame_doubleloop_plot.png")
```
_Output figure_
![issue_frame_doubleloop_plot](https://user-images.githubusercontent.com/94163266/219978530-ef873618-d067-4e27-8292-c83c6f82666c.png)


**(1c) `pygmt.Figure.set_panel:` + using `pygmt.Figure.basemap` WITH setting the `frame` parameter again**
_Example code_
```python
import pygmt

nrows = 1
ncols = 2

fig = pygmt.Figure()

with fig.subplot(
    nrows=nrows,
    ncols=ncols,
    figsize=("10c", "5c"),
    sharex="b+llabel_x",
    sharey="l+llabel_y",
    frame="a1f0.5g2",
):

    for ip in range(nrows*ncols):

        with fig.set_panel(panel=ip):
            
            fig.basemap(
                region=[0, 10, 0, 10],
                projection="X?",
                frame="a1f0.5g5", "x+llabel_x", "y+llabel_y",  # >>> CHANGED gridlines, rest the same
            )

fig.show()
# fig.savefig(fname="issue_frame_panelTrue_basemap_frame.png")
```
_Output figure_
![issue_frame_panelTrue_basemap_frame](https://user-images.githubusercontent.com/94163266/219978402-60b2699f-71b8-40cb-90a5-afd956a6ff9f.png)

**Comment by seisman** [comment]
I think this is the side-effect of the following two lines:

https://github.com/GenericMappingTools/pygmt/blob/34fed4d72d67b2b7f3a311004993f311627ab87c/pygmt/src/basemap.py#L89-L90

@yvonnefroehlich Could you please try remove these two lines and see if everything works as expected?

**Comment by yvonnefroehlich** [comment]
> I think this is the side-effect of the following two lines:
> 
> https://github.com/GenericMappingTools/pygmt/blob/34fed4d72d67b2b7f3a311004993f311627ab87c/pygmt/src/basemap.py#L89-L90
> 
> @yvonnefroehlich Could you please try remove these two lines and see if everything works as expected?

Thanks for pointing out this @seisman!
Please see PR #2417.

**Comment by seisman** [comment]
Closed by #2417.
